client/src/components/ImageUpload/ImageUpload.tsx

```diff
@@ -0,0 +1,60 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+import Dropzone, { DropzoneState } from 'react-dropzone';
+import Button from '@material-ui/core/Button';
+import Grid from '@material-ui/core/Grid';
+import useStyles from './useStyles';
+import { useState } from 'react';
+import { Input, Typography } from '@material-ui/core';
+
+interface Props {
+  handleUpload: (files: File[]) => void;
+}
+
+const ImageUpload = ({ handleUpload }: Props): JSX.Element => {
+  const classes = useStyles();
+  const [files, setFiles] = useState<File[]>([]);
+
+  const onDrop = (acceptedFiles: File[]) => {
+    setFiles(
+      acceptedFiles.map((file) =>
+        Object.assign(file, {
+          preview: URL.createObjectURL(file),
+        }),
+      ),
+    );
+  };
+
+  const previews = files.map((file: any) => (
+    <Grid className={classes.thumb} key={file.name}>
+      <Grid className={classes.thumbInner}>
+        <img src={file.preview} className={classes.img} />
+      </Grid>
+    </Grid>
+  ));
+
+  return (
+    <Dropzone maxFiles={1} onDrop={onDrop}>
+      {(state: DropzoneState) => (
+        <>
+          <Grid container>
+            <Grid {...state.getRootProps({ className: classes.dropzone })}>
+              {/* @ts-ignore */}
```

Comment:
This issue is caused due to ` <Input inputProps={state.getInputProps()} />` giving errors with the typing so that is why I had to do that. Type 'DropzoneInputProps' is not assignable to type 'InputBaseComponentProps'. The type set by react dropzone does not match the type required by MUI, I have tried to fix this and came up with this solution.

Reply:
![image](https://user-images.githubusercontent.com/55607809/140563048-ffc3cedd-74bf-4554-acbc-06b4bddab3ec.png)